compiler/ilgen/IlBuilder.cpp

```diff
@@ -1450,7 +1450,7 @@ IlBuilder::shiftOpNodeFromNodes(TR::ILOpCodes op,
    {
    TR::DataType leftType = leftNode->getDataType();
    TR::DataType rightType = rightNode->getDataType();
-   TR_ASSERT((leftType == TR::Int32 || leftType == TR::Int64) && rightType == TR::Int32, "shift operation first operand must be 32-bit or 64-bit integer, and shift amount must be 32-bit integer");
+   TR_ASSERT((leftType == TR::Int8 || leftType == TR::Int16 || leftType == TR::Int32 || leftType == TR::Int64) && rightType == TR::Int32, "shift operation first operand must an integer, and shift amount must be 32-bit integer");
```

Comment:
Maybe use `leftType.isIntegral()`?

Also "must **be** an integer" in the assert message.